src/main/java/gov/nasa/jpl/mbee/mdk/mms/actions/GeneratePdfAction.java

```diff
@@ -0,0 +1,167 @@
+        this(DEFAULT_ID, "PDF", view);
+
+    }
+
+    protected File fileSelect(String title, File defaultFile, String approveButtonText, FileNameExtensionFilter filter) {
+        JFileChooser choose = new JFileChooser();
+        if (defaultFile != null) {
+            if (defaultFile.isDirectory()) {
+                choose.setCurrentDirectory(defaultFile);
+            }
+            else {
+                choose.setSelectedFile(defaultFile);
+            }
+        }
+        choose.setDialogTitle(title);
+
+        if (filter != null) {
+            choose.addChoosableFileFilter(filter);
+            choose.setFileFilter(filter);
+        }
+        int retval = choose.showDialog(null, approveButtonText);
+        if (retval == JFileChooser.APPROVE_OPTION) {
+            File selectedFile = choose.getSelectedFile();
+            int overwriteOption;
+            if (selectedFile.exists() && approveButtonText.toLowerCase().compareTo("save") == 0) {
+                overwriteOption = JOptionPane.showConfirmDialog(choose, "The file already exists. Do you want to overwrite it?", "Overwrite Confirmation", JOptionPane.OK_CANCEL_OPTION);
+                if (overwriteOption == JOptionPane.OK_OPTION) {
+                    return selectedFile;
+                }
+                else {
+                    return null;
+                }
+            }
+            else
+            	return selectedFile;
+        }
+        return null;
+    }
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        try {
+            //1. ask for the outputfile (pdf) file
+            File outputPdfFile = askForPdfFile();
+            if (outputPdfFile != null) {
+                //assign docbook xml be created filename based on the pdf file
+                File autoGeneratedDocBookFile = new File(outputPdfFile + ".xml");
+                //2. get stylesheet (xsl) info from options->Environment->MDK
+                if (checkForStyleSheetXmlFile()) {
+                    generate(autoGeneratedDocBookFile, xslDefaultFile, outputPdfFile);
+                }
+            }
+        } catch (Exception ex) {
+            Application.getInstance().getGUILog().log("[ERROR] Failed to generate a PDF." + ExceptionUtils.getStackTrace(ex));
+        }
+    }
+    protected boolean checkForStyleSheetXmlFile() {
+        Application.getInstance().getGUILog().log("[INFO] The default stylesheet can be set at \"Default Docbook to PDF Stylesheet\" in Options -> Environment -> MDK");
+        if (MDKOptionsGroup.getMDKOptions().getDocBookToPDFStyleSheet() != null && !MDKOptionsGroup.getMDKOptions().getDocBookToPDFStyleSheet().trim().isEmpty()) {
+            xslDefaultFile = new File(MDKOptionsGroup.getMDKOptions().getDocBookToPDFStyleSheet().trim());
+        }
+        else {
+            xslDefaultFile = new File(MDKPlugin.getInstance().getDescriptor().getPluginDirectory(), "docbook-xsl-1.79.1" + File.separator + "fo" + File.separator + "mdk-default.xsl");
```

Comment:
Let's not include the version number in the folder name, so it doesn't break on upgrade, i.e. `docbook-xsl-1.79.1` -> `docbook-xsl`. This will also require moving `mdk-default.xml` and renaming (during copy) in Gradle `preDist`.

Reply:
Removed docbook version dependency from the code.